app/src/main/java/com/elmeradrianv/shesafe/fragments/MapFragment.java

```diff
@@ -134,15 +184,85 @@ public void onLocationChanged(Location location) {
+                currentLocation.getLongitude());
+    }
+
+    private double positiveRemainder(double divisor, double dividend) {
+        double quotient = Math.floor(dividend / divisor);
+        return dividend - divisor * quotient;
+    }
+
+    private HashMap<Integer, ParsePolygon> getActualGridSquare() {
+        HashMap<Integer, ArrayList<ParseGeoPoint>> grid = new HashMap<>();
+        //squareSize is a "debug number and it will change depends of the speedVector"
+        double squareSize = 0.005;
+        int numSquareSizeGrid = 3;
+        double cornerLatitudeCC = currentLocation.getLatitude() - positiveRemainder(squareSize, currentLocation.getLatitude());
+        double cornerLongitudeCC = currentLocation.getLongitude() - positiveRemainder(squareSize, currentLocation.getLongitude());
+        //saving the number of points in the grid
+        LatLng[][] gridCorners = new LatLng[numSquareSizeGrid + 1][numSquareSizeGrid + 1];
+        for (int i = 0; i < gridCorners.length; i++) {
+            for (int j = 0; j < gridCorners[0].length; j++) {
+                gridCorners[i][j] = new LatLng(
+                        cornerLatitudeCC - (1 - i) * squareSize,
+                        cornerLongitudeCC - (1 - j) * squareSize
+                );
+            }
+        }
+        //k is nine because the grid has 9 squares
+        for (int k = 0; k < 9; k++) {
+            grid.put(k, new ArrayList<>());
+            PolygonOptions polygonOptions = new PolygonOptions();
+            for (int i = k % numSquareSizeGrid; i < k % numSquareSizeGrid + 2; i++) {
+                //Saving the corners of each square
+                if (i - k % numSquareSizeGrid == 0) {
+
+                    for (int j = k / numSquareSizeGrid; j < k / numSquareSizeGrid + 2; j++) {
+                        grid.get(k).add(new ParseGeoPoint(
+                                gridCorners[i][j].latitude,
+                                gridCorners[i][j].longitude)
+                        );
+                        polygonOptions.add(gridCorners[i][j]);
+                    }
+                } else {
+                    for (int j = k / numSquareSizeGrid + 1; j >= k / numSquareSizeGrid; j--) {
+                        grid.get(k).add(new ParseGeoPoint(
+                                gridCorners[i][j].latitude,
+                                gridCorners[i][j].longitude)
+                        );
+                        polygonOptions.add(gridCorners[i][j]);
+                    }
+                }
+            }
+            polygonOptions.add(gridCorners[k % numSquareSizeGrid][k / numSquareSizeGrid]);
+            Polygon polygon = map.addPolygon(polygonOptions);
+        }
+        return getGridPolygons(grid);
+    }
+
+    private HashMap<Integer, ParsePolygon> getGridPolygons(HashMap<Integer, ArrayList<ParseGeoPoint>> grid) {
+        HashMap<Integer, ParsePolygon> gridPolygons = new HashMap<>();
+        for (int k = 0; k < 9; k++) {
```

Comment:
I'd replace the 9 with the SQUARED_GRID_COUNT (or whatever you want to name the "number of locations in my grid of defined size" constant).